noir/aztec_macros/src/lib.rs

```diff
@@ -252,12 +252,15 @@ fn check_for_storage_definition(module: &SortedModule) -> bool {
     module.types.iter().any(|r#struct| r#struct.name.0.contents == "Storage")
 }
 
-// Check if "compute_note_hash_and_nullifier(Field,Field,Field,[Field; N]) -> [Field; 4]" is defined
+// Check if "compute_note_hash_and_nullifier(AztecAddress,Field,Field,[Field; N]) -> [Field; 4]" is defined
 fn check_for_compute_note_hash_and_nullifier_definition(module: &SortedModule) -> bool {
     module.functions.iter().any(|func| {
         func.def.name.0.contents == "compute_note_hash_and_nullifier"
                 && func.def.parameters.len() == 4
-                && func.def.parameters[0].typ.typ == UnresolvedTypeData::FieldElement
+                && match &func.def.parameters[0].typ.typ {
```

Comment:
More a question really, but do you know why it is `typ.typ` instead of just a single `typ`?

Reply:
The first "typ" has `UnresolvedType` and the second one is `UnresolvedTypeData`. It looks like a bit of a messy naming.

This is more of a question for someone from noir team (maybe @kevaundray ?).